### ISSUE TITLE
Bug: `no-fallthough` doesn't account for `process.exit()`

### ISSUE THREAD

**Issue by RedGuy12** [eslint/eslint#17466]
### Environment

Node version: 18.12.1
npm version: 8.13.2
ESLint version: 8.47.0
Operating System: Windows

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
	env: { es2021: true, node: true },
	extends: ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
	parser: "@typescript-eslint/parser",
	parserOptions: { ecmaVersion: "latest", sourceType: "module" },
	plugins: ["@typescript-eslint"],
};
```
</details>

```js
switch (variable) {
	case 1: {
		// do stuff
		process.exit(1);
	}
	case 2: {
		// do stuff
		break;
	}
	// ...
}
```

### What did you expect to happen?

No error, since `process.exit()` prevents `case 2` from executing

### What actually happened?

```
...\file.ts
  6:2  error  Expected a 'break' statement before 'case'  no-fallthrough
```

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWZhbGx0aHJvdWdoOiBcImVycm9yXCIgKi9cbnN3aXRjaCAodmFyaWFibGUpIHtcblx0Y2FzZSAxOiB7XG5cdFx0Ly8gZG8gc3R1ZmZcblx0XHRwcm9jZXNzLmV4aXQoMSk7XG5cdH1cblx0Y2FzZSAyOiB7XG5cdFx0Ly8gZG8gc3R1ZmZcblx0XHRicmVhaztcblx0fVxuXHQvLyAuLi5cbn0iLCJvcHRpb25zIjp7ImVudiI6eyJub2RlIjp0cnVlLCJlczYiOnRydWV9LCJydWxlcyI6e30sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fQ==

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Rec0iL99** [comment]
Hi @RedGuy12, thanks for the issue. As stated in #17222, core rules don't have type information to know if a function returns, which in this case is `process.exit()`. 

You might want to look into https://github.com/typescript-eslint/typescript-eslint/issues/3455 which is currently accepting prs and will help in cases like this.
